cgi-bin/CATS/ApiJudge.pm

```diff
@@ -171,9 +155,19 @@ sub select_request {
     my ($p) = @_;
     $p->{supported_DEs} or return print_json({ error => 'bad request' });
 
-    my ($jid) = $dbh->selectrow_array(q~
-        SELECT J.id FROM judges J INNER JOIN accounts A ON J.account_id = A.id WHERE A.sid = ?~, undef,
+    my ($was_pinged, $lock_counter, $jid, $time_since_alive) = $dbh->selectrow_array(q~
+        SELECT 1 - J.is_alive, J.lock_counter, J.id, CURRENT_TIMESTAMP - J.alive_date
+        FROM judges J INNER JOIN accounts A ON J.account_id = A.id WHERE A.sid = ?~, undef,
         $sid);
+
+    $dbh->do(q~
+        UPDATE judges SET is_alive = 1, alive_date = CURRENT_TIMESTAMP WHERE id = ?~, undef,
+        $jid) if $was_pinged || $time_since_alive > $CATS::Config::judge_alive_interval / 24;
+    $dbh->commit;
+
+    return print_json({ lock_counter => $lock_counter, was_pinged => $was_pinged, request => undef })
+        if $lock_counter;
```

Comment:
This seems a little duplicated.
How about `my $response = { request => undef }`, then `($response->{was_pinged}...)` and finally
return `print_json($response)`.